### ISSUE TITLE
Websites favicons not blocked if behind-the-scene requests are whitelisted

### ISSUE THREAD

**Issue by acrobat1** [gorhill/uBlock#1732]
I'm seeing that websites favicons aren't actually blocked in Firefox even with a valid dynamic/static  filter, if behind-the-scene requests are allowed (_as it is by default_) .

For instance, a filter like this one 
`||github.com/favicon.ico$domain=behind-the-scene,important`
 or even more generic such as
 `||github.com/favicon.ico$important`
will do nothing if behind-the-scene requests are allowed in the Whitelist tab, but will definitely work as soon as behind-the-scene is un-whitelisted.

When commenting out behind-the-scene in the Whitelist tab, favicons are indeed totally blocked as expected, but with the default allow rule all filters specific to favicons don't seem to work, as favicons requests appear in the logger every now and then: is that the expected behaviour, meaning that uBlock can't really apply such filters as long as 
behind-the-scene requests are generally whitelisted?
## Steps for anyone to reproduce the issue

Set a dynamic or static filter in order to block a website favicon: using the logger, select a favicon for a website, then try to block it using either a dynamic filter or a static filter, like the example above - leaving behind-the-scene whitelisted (per default settings) in the Whitelist tab.
You will see that these filters won't really work, as the favicon which should be now blocked is  actually  going to pop up  again in the log sooner or later (maybe after clearing the cache or forcing a reload of the page bypassing the cache) .
Then comment out behind-the-scene in the Whitelist tab, and notice that favicons for which a filter exists will now be properly blocked.
## Your settings
- Browser/version: 
  Firefox 47 release
  Mozilla/5.0 (X11; Linux x86_64; rv:47.0) Gecko/20100101 Firefox/47.0
- uBlock Origin version: 
  uBlock Origin v1.7.0

**Comment by gorhill** [comment]
From what I can gather, this is a Firefox quirk -- related issues:
- <https://bugzilla.mozilla.org/show_bug.cgi?id=437014>.
- <https://bugzilla.mozilla.org/show_bug.cgi?id=1196013>.

Sometimes favicon-based network do not go through `shouldLoad` -- this will cause uBO to be unable to associate the network request with a specific tab, hence it will be categorized as behind-the-scene. Here is the issue illustrated using ABP:

![a](https://cloud.githubusercontent.com/assets/585534/16178214/1428d0b0-3610-11e6-91b5-0f53958d1d8d.png)

As can be seen, ABP reports the favicon as being blocked -- quite probably because it indeed was in `shouldLoad` handler, but the browser console shows the favicon request suceeding, quite probably because another one was made without going through `shouldLoad` handler.

In short, a browser issue -- unsure whether by design or not. The solution a uBO user is to disable the whitelisting of behind-the-scene scope (and assume the consequences of such whitelisting).

**Comment by acrobat1** [comment]
Thanks for explaining this, I assume as long as Firefox folks won't fix this (_if they' ll ever considering fixing it_), there's nothing uBlock can do-
that's  why I've also opened this feature request [Add a quick toggle for behind-the-scene requests in Advanced mode](https://github.com/gorhill/uBlock/issues/1734) .
